File: src/test/java/com/uapp_llc/test/model/wrapper/task/TaskWrapper.java
package com.uapp_llc.test.model.wrapper.task;

import com.uapp_llc.model.Task;
import com.uapp_llc.test.model.mutator.TaskMutators;
import com.uapp_llc.test.model.wrapper.AbstractWrapper;

abstract class TaskWrapper extends AbstractWrapper<Task> {

  protected TaskWrapper(String name, String description, Integer index) {
    super(new Task());
    super
        .with(TaskMutators.name(name))
        .with(TaskMutators.description(description))
        .with(TaskMutators.index(index));
  }

}

File: src/test/kotlin/com/uapp_llc/test/model/wrapper/task/TaskWrapper.kt
package com.uapp_llc.test.model.wrapper.task

import java.time.ZonedDateTime
import com.uapp_llc.model.Task
import com.uapp_llc.test.model.wrapper.ModelWrapper

abstract class TaskWrapper constructor(

    name: String,
    description: String,
    index: Int,
    createdAt: ZonedDateTime = ZonedDateTime.now()

) : ModelWrapper<Task>(Task(null, createdAt, name, description, index))
